enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/MiscAcceptanceTest.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2002-2018 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.internal.cypher.acceptance
+
+import org.neo4j.cypher.{ExecutionEngineFunSuite, NewPlannerTestSupport, SyntaxException}
+
+class MiscAcceptanceTest extends ExecutionEngineFunSuite with NewPlannerTestSupport {
+
+  test("should be able to use long values for LIMIT in interpreted runtime") {
+    val a = createNode()
+    val b = createNode()
+
+    val limit: Long = Int.MaxValue + 1l
+    // If we would use Ints for storing the limit, then we would end up with "limit 0"
+    // thus, if we actually return the two nodes, then it proves that we used a long
+    val query = "CYPHER runtime = interpreted MATCH (n) RETURN n LIMIT " + limit
+    val result = innerExecute(query)
+    result.toList should equal(List(Map("n" -> a), Map("n" -> b)))
+  }
+
+  test("should not be able to use more then Long.MaxValue for LIMIT in interpreted runtime") {
+    val expectedErrorMessage = "Invalid input for LIMIT. Either the string does not have the appropriate format or the provided number is bigger then 2^63-1"
```

Comment:
We usually use either `intercept` to check errors of if you want to be really fancy you can do

```
val error = the [SyntaxException] thrownBy innerExecute(query)
error.getMessage() should equal(expectedErrorMessage)
```